Parser/Types/TypeAnalyze.cs

```diff
@@ -23,15 +23,32 @@ public override NType Visit (NProgram p)
    }
 
    public override NType Visit (NDeclarations d) {
-      Visit (d.Vars); return Visit (d.Funcs);
+      Visit (d.Consts);  Visit (d.Vars); return Visit (d.Funcs);
+   }
+
+   public override NType Visit (NConstDecl c) {
+      mSymbols.Consts.Add (c);
+      ((NLiteral)c.Expr).Accept (this);
+      return c.Expr.Type;
    }
 
    public override NType Visit (NVarDecl d) {
+      // Check multiple varialbes declared in the same scope.
+      var node = mSymbols.Find (d.Name.Text);
+      if (node != null) {
+         string? txt = node is NConstDecl ? $"{((NConstDecl)node).Expr.Type}" : $"{((NVarDecl)node).Type}";
+         throw new ParseException (d.Name, $"A variable named '{d.Name.Text}' of type '{txt}' is already defined in this scope");

```

Comment:
1. What if node is an NFnDecl?
2. Compute the text type more elegantly using a switch expression, and avoid the typecasts:
```
NType type = node switch {
   NConstDecl c => c.Expr.Type, 
   NVarDecl v => v.Type,
   NFnDecl f => f.Return, 
   _ => throw new NotImplementedException ()
```
}